pandas/core/frame.py

```diff
@@ -2866,7 +2866,12 @@ def __getitem__(self, key):
         if is_single_key:
             if self.columns.nlevels > 1:
                 return self._getitem_multilevel(key)
+
             indexer = self.columns.get_loc(key)
+            if isinstance(self.columns, ABCIntervalIndex):
```

Comment:
rather would not inject specific index logic here

you can make this not a single key (so it takes a different branch)

or define one of the _convert_to_index routines which dispatch to the index type